src/main/java/it/smartcommunitylab/aac/saml/provider/SamlAuthenticationProvider.java

```diff
@@ -227,6 +240,18 @@ protected SamlUserAuthenticatedPrincipal createUserPrincipal(Object principal) {
         // we still don't have userId
         String userId = null;
 
+        // subjectId is optionally evaluated from an attribute defined in provider configurations
+        String subjectId = samlDetails.getName();
```

Comment:
move the reading of subject back to the previous place, before the `getName`. It is the first thing we need to read.